examples/integration-tutorials/transforming-message-content/message_transformation_service/message_transformation.bal

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/log;
+
+//Define endpoint for the backen service.
+http:Client hospitalEP = new("http://localhost:9090");
+
+//Constants for request paths.
+const GRAND_OSK_EP_PATH = "/grandoaks/categories/";
+const CLEMENCY_EP_PATH = "/clemency/categories/";
+const PINE_VALLEY_EP_PATH = "/pinevalley/categories/";
+
+//Constant for error code.
+const string ERROR_CODE = "Sample Error";
+
+@http:ServiceConfig {
+    basePath: "/healthcare"
+}
+service healthcareService on new http:Listener(9091) {
+    @http:ResourceConfig {
+        methods: ["POST"],
+        path: "/categories/{category}/reserve"
+    }
+    resource function makeReservation(http:Caller caller, http:Request request, string category) {
+        //Extract payload from the request.
+        var requestPayload = request.getJsonPayload();
+        json modifiedPayload = {};
+        string requestPath = "";
+
+        //Define new response. 
+        http:Response|error backendResponse = new();
+
+        if (requestPayload is json) {
+            //Get hospital name.
+            string hospitalName = requestPayload.hospital.toString();
+            //Transform the payload into the format which is required by the backend service.
+            modifiedPayload = {
+                "patient": {
+                    "name": requestPayload.name,
+                    "dob": requestPayload.dob,
+                    "ssn": requestPayload.ssn,
+                    "address": requestPayload.address,
+                    "phone": requestPayload.phone,
+                    "email": requestPayload.email,
+                    "cardNo": requestPayload.cardNo
+                },
+                "doctor": requestPayload.doctor,
+                "hospital": hospitalName,
+                "appointment_date": requestPayload.appointment_date
+            };
```

Comment:
What about the else path? We need to handle the error. In addition, even if respond back the client the execution flow continues regardless of that. So we need to keep everything within the if block. Maybe to improve readability we can move code segments into their own functions.